spec/features/posts_index_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe 'post#index', type: :feature do
+  describe 'post page' do
+    before(:each) do
+      @user = User.create!(name: 'Amare23', photo: 'my-photo2', bio: 'Microverse student22')
+      @post = Post.create!(title: 'Integ test22', text: 'differs much in using matchers...', author_id: @user.id)
+      @comment = Comment.create!(text: 'I need help!', author: @user, post: @post)
+      Like.create!(author_id: @user.id, post_id: @post.id)
+
+      visit(user_posts_path(@user))
+    end
+
+    it " should display users' username" do
+      expect(page).to have_content('Amare')
+    end
+
+    it 'should display the number of posts for the user to be 1' do
+      expect(page).to have_content('Number of posts: 1')
+    end
+
+    it 'should not display the number of posts for the user to be 2' do
+      expect(page).to_not have_content('Number of posts: 2')
+    end
+
+    it "\nshould display posts author name" do
+      expect(page).to have_content(@post.author.name)
+    end
+
+    it "\nshould display posts title" do
+      expect(page).to have_content(@post.title)
+    end
+
+    it "\nshould display the body of the post text" do
+      expect(page).to have_content 'differs much in using'
+    end
+
+    it "\nshould display the comment on this post" do
+      expect(page).to have_content('I need help!')
+    end
+
+    it "\nshould display the comments counter for the post to be 1" do
+      expect(page).to have_content("Comments: #{@post.comments_counter}")
+    end
+
+    it "\nshould display the likes counter for the post to be 1" do
+      expect(page).to have_content('Likes: 1')
+    end
+
+    it "\nshould display post details when the user clicks title of the post link" do
```

Comment:
- [ ] For the post index page you should have tests that assert you can see the user's profile picture.
![image](https://user-images.githubusercontent.com/62705947/217493457-f05a9482-20c8-4ee3-b490-461b43b61388.png)

Reply:
- [ ] In the posts index page you should have tests that assert you can see a section for pagination if there are more posts than fit on the view.
![image](https://user-images.githubusercontent.com/62705947/217493984-22b6fe65-8135-4246-a615-6c5f247dec9c.png)